src/frontend/screens/AudioPlaybackSaved.tsx

```diff
@@ -0,0 +1,198 @@
+import * as React from 'react';
+import {
+  AudioStyles,
+  SIDE_ICON_BUTTON_WIDTH,
+} from '../sharedComponents/AudioStyles';
+import {ScreenContentWithDock} from '../sharedComponents/ScreenContentWithDock';
+import {useAudioPlayback} from '../hooks/useAudioPlayback';
+import {Duration} from 'luxon';
+import {View, Text, Pressable, TouchableOpacity} from 'react-native';
+import {Bar} from 'react-native-progress';
+import {WHITE, MEDIUM_GREY} from '../lib/styles';
+import {ErrorBottomSheet} from '../sharedComponents/ErrorBottomSheet';
+import {defineMessages, useIntl} from 'react-intl';
+import {HeaderText} from '../sharedComponents/Text/HeaderText';
+import {NativeRootNavigationProps} from '../sharedTypes/navigation';
+import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
+import PlayArrow from '../images/PlayArrow.svg';
+import {UIActivityIndicator} from 'react-native-indicators';
+import Share from 'react-native-share';
+import * as FileSystem from 'expo-file-system';
+
+const m = defineMessages({
+  description: {
+    id: 'screens.AudioPlaybackUnsaved.description',
+    defaultMessage: 'Total length: {length}',
+  },
+});
+export const AudioPlaybackSaved = ({
+  route,
+  navigation,
+}: NativeRootNavigationProps<'AudioPlaybackSaved'>) => {
+  const uri = route.params.uri;
+  const {
+    duration,
+    currentPosition,
+    isPlaying,
+    stopPlayback,
+    startPlayback,
+    error,
+    clearError,
+  } = useAudioPlayback(uri);
+  const {formatMessage} = useIntl();
+
+  const progress = currentPosition / duration;
+
+  const [localUri, setLocalUri] = React.useState<string | null>(null);
+  const [shareLoading, setShareLoading] = React.useState(false);
+  const [shareError, setShareError] = React.useState<Error | null>(null);
+
+  const handleShare = React.useCallback(async () => {
+    setShareLoading(true);
+    try {
+      let fileUri = localUri;
+
+      if (!fileUri) {
+        const tempFileName = `audio_${Date.now()}.m4a`;
+        const localFilePath = `${FileSystem.cacheDirectory}${tempFileName}`;
+        const {uri: downloadedUri} = await FileSystem.downloadAsync(
+          uri,
+          localFilePath,
+        );
+
+        fileUri = downloadedUri;
+        setLocalUri(downloadedUri);
+      }
+
+      await Share.open({url: fileUri, failOnCancel: false});
+    } catch (err) {
+      setShareError(err as Error);
```

Comment:
slightly more resilient error handling:

```suggestion
      if (err instanceof Error) {
        setShareError(err)
      } else {
        // Replace message with something more useful for debugging
        setShareError(new Error('Unable to share audio file'))
      }
```

Reply:
https://github.com/digidem/comapeo-mobile/pull/958/commits/5d09e7e5176e12da23a844e045b7851354ac1406